web-api/src/error/common.rs

```diff
@@ -338,4 +339,32 @@ impl_from_std_error!(
     std::io::Error,
     tokio::task::JoinError,
     serde_json::Error,
+    xayn_snippet_extractor::Error,
 );
+
+impl ApplicationError for PoolError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            PoolError::Timeout => StatusCode::SERVICE_UNAVAILABLE,
+            PoolError::Closed | PoolError::NoRuntimeSpecified => StatusCode::INTERNAL_SERVER_ERROR,
+        }
+    }
+
+    fn kind(&self) -> &str {
+        match self {
+            PoolError::Timeout => "ServiceOverloaded",
```

Comment:
I am not convinced about exposing "ServiceOverloaded"

Reply:
can you explain why, an overload of the service is the only way you can normally run into this timeout  (it's a timeout  on retrieving a extractor)

Reply:
As far as I remember we also did discuss exposing errors cause by elasticsearch being overloaded due to too many parallel ingestion as `503 Service Unavailable` in the future, which is a similar case.